api/diagnostics.go

```diff
@@ -812,8 +812,11 @@ func (j *DiagnosticsJob) findLocalBundle() (bundles []string, err error) {
 func matchRequestedNodes(requestedNodes []string, masterNodes, agentNodes []dcos.Node) ([]dcos.Node, error) {
 	var matchedNodes []dcos.Node
 	clusterNodes := append(masterNodes, agentNodes...)
-	if len(requestedNodes) == 0 || len(clusterNodes) == 0 {
-		return matchedNodes, errors.New("Cannot match requested nodes to clusterNodes")
+	if len(requestedNodes) == 0 {
+		return matchedNodes, errors.New("no nodes were requested")
```

Comment:
I think this would be more reasonable here:
```suggestion
		return nil, errors.New("no nodes were requested")
```